app/src/main/java/fraktal/io/android/demo/workers/profile/WorkerScreen.kt

```diff
@@ -0,0 +1,435 @@
+                    keyboardOptions = KeyboardOptions(imeAction = ImeAction.Next, keyboardType = KeyboardType.Email),
+                    onStateChange = {
+                        mutableState = mutableState.copy(email = it)
+                    }
+                )
+                DropDownMenu(item = mutableState.gender, variants = genders, placeHolder = "Gender") {
+                    mutableState = mutableState.copy(gender = mutableState.gender.putNewData(it))
+                }
+                DropDownMenu(item = mutableState.position, variants = positions, placeHolder = "Position") {
+                    mutableState = mutableState.copy(position = mutableState.position.putNewData(it))
+                }
+                InputTextFiled(
+                    mutableState = mutableState.phoneNumber,
+                    placeHolder = "Enter phone number",
+                    maxSymbols = 15,
+                    keyboardActions = KeyboardActions(
+                        onNext = {
+                            focusManager.moveFocus(FocusDirection.Down)
+                        }
+                    ),
+                    keyboardOptions = KeyboardOptions(imeAction = ImeAction.Next, keyboardType = KeyboardType.Phone),
+                    onStateChange = {
+                        mutableState = mutableState.copy(phoneNumber = it)
+                    }
+                )
+                InputTextFiled(
+                    mutableState = mutableState.date,
+                    placeHolder = "Enter date of birth",
+                    keyboardActions = KeyboardActions(
+                        onNext = {
+                            focusManager.moveFocus(FocusDirection.Down)
+                        }
+                    ),
+                    maxSymbols = 8,
+                    keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done, keyboardType = KeyboardType.Number),
+                    supportText = "dd/mm/yyyy",
+                    visualTransformation = DateTransformation,
+                    onStateChange = {
+                        mutableState = mutableState.copy(date = it)
+                    }
+                )
+            }
+            Button(
+                onClick = {
+                    onButtonClick(
+                        prepareCommandIfValid(mutableState) {
+                            mutableState = it
+                        }
+                    )
+                },
+                enabled = mutableState.isEditMode.not() or changedState,
+                modifier = Modifier
+                    .align(BottomCenter)
+                    .padding(bottom = 8.dp)
+            ) {
+                Text(
+                    text = if (mutableState.isEditMode) "Edit" else "Save",
+                    modifier = Modifier.padding(horizontal = 32.dp)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun <T> InputTextFiled(
+    mutableState: InputItem<T>,
```

Comment:
about 1.
I've been thinking a lot about where to put it, and I've come to 2 important conclusions.

1. the main logic of the application works only with valid data. Formatting and validation look like auxiliary functions. by placing them in deicider, we will not need to complicate the main algorithm of the feature
2. it will be easier and more correct to store this ui-logic on a compase 
https://developer.android.com/develop/ui/compose/state-hoisting